driver/modern_bpf/programs/attached/dispatchers/syscall_exit.bpf.c

```diff
@@ -87,6 +87,44 @@ int BPF_PROG(sys_exit,
 		return 0;
 	}
 
+	struct filter_map_entry *filter = maps__get_filter_for_syscall_num(syscall_id);
+	if (filter != NULL) {
+		const void *name_pointer = (const void *) extract__syscall_argument(regs, 1);
+		int limit;
+		if (filter->num_prefixes > 12) { // 12 seems to be the limit for ARM with the current implementation
+			limit = 12;
+		} else {
+			limit = filter->num_prefixes; 
+		}
+		for (int filter_num = 0; filter_num < limit; filter_num++) {
+			char open_path_prefix[32];
+			bpf_probe_read_user_str(open_path_prefix, 32, name_pointer);
+			int match = 1;
+			if (filter->prefixes[filter_num][0] == '\0') // filter prefix is null, this means we've hit the end of the list
+			{
+				break;
```

Comment:
From PR review: can probably remove this `break` in favor of an assignment and check on `match`